// melange-webapi/Webapi/Canvas/Webapi__Canvas__WebGl.js
// Generated by Melange


const _DEPTH_BUFFER_BIT = 256;

const _STENCIL_BUFFER_BIT = 1024;

const _COLOR_BUFFER_BIT = 16384;

const _POINTS = 0;

const _LINES = 1;

const _LINE_LOOP = 2;

const _LINE_STRIP = 3;

const _TRIANGLES = 4;

const _TRIANGLE_STRIP = 5;

const _TRIANGLE_FAN = 6;

const _CULL_FACE = 2884;

const _BLEND = 3042;

const _DITHER = 3024;

const _STENCIL_TEST = 2960;

const _DEPTH_TEST = 2929;

const _SCISSOR_TEST = 3089;

const _POLYGON_OFFSET_FILL = 32823;

const _SAMPLE_ALPHA_TO_COVERAGE = 32926;

const _SAMPLE_COVERAGE = 32928;

const _ZERO = 0;

const _ONE = 1;

const _SRC_COLOR = 768;

const _ONE_MINUS_SRC_COLOR = 769;

const _SRC_ALPHA = 770;

const _ONE_MINUS_SRC_ALPHA = 771;

const _DST_ALPHA = 772;

const _ONE_MINUS_DST_ALPHA = 773;

const _BYTE = 5120;

const _UNSIGNED_BYTE = 5121;

const _SHORT = 5122;

const _UNSIGNED_SHORT = 5123;

const _INT = 5124;

const _UNSIGNED_INT = 5125;

const _FLOAT = 5126;

const _FRONT = 1028;

const _BACK = 1029;

const _FRONT_AND_BACK = 1032;

const _FRAGMENT_SHADER = 35632;

const _VERTEX_SHADER = 35633;

const _ARRAY_BUFFER = 34962;

const _ELEMENT_ARRAY_BUFFER = 34963;

const _ARRAY_BUFFER_BINDING = 34964;

const _ELEMENT_ARRAY_BUFFER_BINDING = 34965;

const _STREAM_DRAW = 35040;

const _STATIC_DRAW = 35044;

const _DYNAMIC_DRAW = 35048;

export {
  _DEPTH_BUFFER_BIT,
  _STENCIL_BUFFER_BIT,
  _COLOR_BUFFER_BIT,
  _POINTS,
  _LINES,
  _LINE_LOOP,
  _LINE_STRIP,
  _TRIANGLES,
  _TRIANGLE_STRIP,
  _TRIANGLE_FAN,
  _CULL_FACE,
  _BLEND,
  _DITHER,
  _STENCIL_TEST,
  _DEPTH_TEST,
  _SCISSOR_TEST,
  _POLYGON_OFFSET_FILL,
  _SAMPLE_ALPHA_TO_COVERAGE,
  _SAMPLE_COVERAGE,
  _ZERO,
  _ONE,
  _SRC_COLOR,
  _ONE_MINUS_SRC_COLOR,
  _SRC_ALPHA,
  _ONE_MINUS_SRC_ALPHA,
  _DST_ALPHA,
  _ONE_MINUS_DST_ALPHA,
  _BYTE,
  _UNSIGNED_BYTE,
  _SHORT,
  _UNSIGNED_SHORT,
  _INT,
  _UNSIGNED_INT,
  _FLOAT,
  _FRONT,
  _BACK,
  _FRONT_AND_BACK,
  _FRAGMENT_SHADER,
  _VERTEX_SHADER,
  _ARRAY_BUFFER,
  _ELEMENT_ARRAY_BUFFER,
  _ARRAY_BUFFER_BINDING,
  _ELEMENT_ARRAY_BUFFER_BINDING,
  _STREAM_DRAW,
  _STATIC_DRAW,
  _DYNAMIC_DRAW,
}
/* No side effect */
