// melange-webapi/Webapi/Webapi__Dom.js
// Generated by Melange

import * as Webapi__Dom__Types from "./Dom/Webapi__Dom__Types.js";

const encodeCompareHow = Webapi__Dom__Types.encodeCompareHow;

const decodeCompareResult = Webapi__Dom__Types.decodeCompareResult;

const decodeCompatMode = Webapi__Dom__Types.decodeCompatMode;

const encodeContentEditable = Webapi__Dom__Types.encodeContentEditable;

const decodeContentEditable = Webapi__Dom__Types.decodeContentEditable;

const decodeDeltaMode = Webapi__Dom__Types.decodeDeltaMode;

const encodeDesignMode = Webapi__Dom__Types.encodeDesignMode;

const decodeDesignMode = Webapi__Dom__Types.decodeDesignMode;

const encodeDir = Webapi__Dom__Types.encodeDir;

const decodeDir = Webapi__Dom__Types.decodeDir;

const EventPhase = Webapi__Dom__Types.EventPhase;

const encodeFilterAction = Webapi__Dom__Types.encodeFilterAction;

const encodeInsertPosition = Webapi__Dom__Types.encodeInsertPosition;

const encodeModifierKey = Webapi__Dom__Types.encodeModifierKey;

const decodeNodeType = Webapi__Dom__Types.decodeNodeType;

const decodePointerType = Webapi__Dom__Types.decodePointerType;

const decodeReadyState = Webapi__Dom__Types.decodeReadyState;

const decodeShadowRootMode = Webapi__Dom__Types.decodeShadowRootMode;

const decodeVisibilityState = Webapi__Dom__Types.decodeVisibilityState;

const WhatToShow = Webapi__Dom__Types.WhatToShow;

export {
  encodeCompareHow,
  decodeCompareResult,
  decodeCompatMode,
  encodeContentEditable,
  decodeContentEditable,
  decodeDeltaMode,
  encodeDesignMode,
  decodeDesignMode,
  encodeDir,
  decodeDir,
  EventPhase,
  encodeFilterAction,
  encodeInsertPosition,
  encodeModifierKey,
  decodeNodeType,
  decodePointerType,
  decodeReadyState,
  decodeShadowRootMode,
  decodeVisibilityState,
  WhatToShow,
}
/* No side effect */
