// melange-webapi/Webapi/Dom/Webapi__Dom__HtmlImageElement.js
// Generated by Melange

import * as Js__Js_null from "melange.js/js_null.js";
import * as Webapi__Dom__Element from "./Webapi__Dom__Element.js";
import * as Webapi__Dom__EventTarget from "./Webapi__Dom__EventTarget.js";
import * as Webapi__Dom__HtmlElement from "./Webapi__Dom__HtmlElement.js";
import * as Webapi__Dom__Node from "./Webapi__Dom__Node.js";

function setCrossOrigin(self, value) {
  self.crossOrigin = Js__Js_null.fromOption(value);
}

const include = Webapi__Dom__Node.Impl({});

Webapi__Dom__EventTarget.Impl({});

const include$1 = Webapi__Dom__Element.Impl({});

const include$2 = Webapi__Dom__HtmlElement.Impl({});

const nodeType = include.nodeType;

const asHtmlElement = include$1.asHtmlElement;

const ofNode = include$1.ofNode;

const insertAdjacentElement = include$1.insertAdjacentElement;

const insertAdjacentHTML = include$1.insertAdjacentHTML;

const insertAdjacentText = include$1.insertAdjacentText;

const ofElement = include$2.ofElement;

const contentEditable = include$2.contentEditable;

const setContentEditable = include$2.setContentEditable;

const dir = include$2.dir;

const setDir = include$2.setDir;

export {
  setCrossOrigin,
  nodeType,
  asHtmlElement,
  ofNode,
  insertAdjacentElement,
  insertAdjacentHTML,
  insertAdjacentText,
  ofElement,
  contentEditable,
  setContentEditable,
  dir,
  setDir,
}
/* include Not a pure module */
