// melange-webapi/Webapi/Dom/Webapi__Dom__HtmlDocument.js
// Generated by Melange

import * as Js__Js_null from "melange.js/js_null.js";
import * as Webapi__Dom__Document from "./Webapi__Dom__Document.js";
import * as Webapi__Dom__EventTarget from "./Webapi__Dom__EventTarget.js";
import * as Webapi__Dom__GlobalEventHandlers from "./Webapi__Dom__GlobalEventHandlers.js";
import * as Webapi__Dom__Node from "./Webapi__Dom__Node.js";
import * as Webapi__Dom__Types from "./Webapi__Dom__Types.js";

function Impl(T) {
  const designMode = function (self) {
    return Webapi__Dom__Types.decodeDesignMode(self.designMode);
  };
  const setDesignMode = function (self, value) {
    self.designMode = Webapi__Dom__Types.encodeDesignMode(value);
  };
  const dir = function (self) {
    return Webapi__Dom__Types.decodeDir(self.dir);
  };
  const setDir = function (self, value) {
    self.dir = Webapi__Dom__Types.encodeDir(value);
  };
  const readyState = function (self) {
    return Webapi__Dom__Types.decodeReadyState(self.readyState);
  };
  const execCommand = function (command, show, value, self) {
    return self.execCommand(command, show, Js__Js_null.fromOption(value));
  };
  return {
    designMode: designMode,
    setDesignMode: setDesignMode,
    dir: dir,
    setDir: setDir,
    readyState: readyState,
    execCommand: execCommand
  };
}

const include = Webapi__Dom__Node.Impl({});

Webapi__Dom__EventTarget.Impl({});

Webapi__Dom__GlobalEventHandlers.Impl({});

const include$1 = Webapi__Dom__Document.Impl({});

function designMode(self) {
  return Webapi__Dom__Types.decodeDesignMode(self.designMode);
}

function setDesignMode(self, value) {
  self.designMode = Webapi__Dom__Types.encodeDesignMode(value);
}

function dir(self) {
  return Webapi__Dom__Types.decodeDir(self.dir);
}

function setDir(self, value) {
  self.dir = Webapi__Dom__Types.encodeDir(value);
}

function readyState(self) {
  return Webapi__Dom__Types.decodeReadyState(self.readyState);
}

function execCommand(command, show, value, self) {
  return self.execCommand(command, show, Js__Js_null.fromOption(value));
}

const nodeType = include.nodeType;

const asHtmlDocument = include$1.asHtmlDocument;

const ofNode = include$1.ofNode;

const compatMode = include$1.compatMode;

const visibilityState = include$1.visibilityState;

export {
  Impl,
  nodeType,
  asHtmlDocument,
  ofNode,
  compatMode,
  visibilityState,
  designMode,
  setDesignMode,
  dir,
  setDir,
  readyState,
  execCommand,
}
/* include Not a pure module */
