// melange-webapi/Webapi/Dom/Webapi__Dom__HtmlElement.js
// Generated by Melange

import * as Caml_option from "melange.js/caml_option.js";
import * as Webapi__Dom__Element from "./Webapi__Dom__Element.js";
import * as Webapi__Dom__EventTarget from "./Webapi__Dom__EventTarget.js";
import * as Webapi__Dom__GlobalEventHandlers from "./Webapi__Dom__GlobalEventHandlers.js";
import * as Webapi__Dom__Node from "./Webapi__Dom__Node.js";
import * as Webapi__Dom__Types from "./Webapi__Dom__Types.js";

function Impl(T) {
  const ofElement = (function (element) {
      // BEWARE: Assumes "contentEditable" uniquely identifies an HTMLELement
      return element.contentEditable !== undefined ?  element : null;
    }
  );
  const ofElement$1 = function (self) {
    return Caml_option.null_to_opt(ofElement(self));
  };
  const contentEditable = function (self) {
    return Webapi__Dom__Types.decodeContentEditable(self.contentEditable);
  };
  const setContentEditable = function (self, value) {
    self.contentEditable = Webapi__Dom__Types.encodeContentEditable(value);
  };
  const dir = function (self) {
    return Webapi__Dom__Types.decodeDir(self.dir);
  };
  const setDir = function (self, value) {
    self.dir = Webapi__Dom__Types.encodeDir(value);
  };
  return {
    ofElement: ofElement$1,
    contentEditable: contentEditable,
    setContentEditable: setContentEditable,
    dir: dir,
    setDir: setDir
  };
}

const include = Webapi__Dom__Node.Impl({});

Webapi__Dom__EventTarget.Impl({});

Webapi__Dom__GlobalEventHandlers.Impl({});

const include$1 = Webapi__Dom__Element.Impl({});

const ofElement = (function (element) {
      // BEWARE: Assumes "contentEditable" uniquely identifies an HTMLELement
      return element.contentEditable !== undefined ?  element : null;
    }
);

function ofElement$1(self) {
  return Caml_option.null_to_opt(ofElement(self));
}

function contentEditable(self) {
  return Webapi__Dom__Types.decodeContentEditable(self.contentEditable);
}

function setContentEditable(self, value) {
  self.contentEditable = Webapi__Dom__Types.encodeContentEditable(value);
}

function dir(self) {
  return Webapi__Dom__Types.decodeDir(self.dir);
}

function setDir(self, value) {
  self.dir = Webapi__Dom__Types.encodeDir(value);
}

const nodeType = include.nodeType;

const asHtmlElement = include$1.asHtmlElement;

const ofNode = include$1.ofNode;

const insertAdjacentElement = include$1.insertAdjacentElement;

const insertAdjacentHTML = include$1.insertAdjacentHTML;

const insertAdjacentText = include$1.insertAdjacentText;

export {
  Impl,
  nodeType,
  asHtmlElement,
  ofNode,
  insertAdjacentElement,
  insertAdjacentHTML,
  insertAdjacentText,
  ofElement$1 as ofElement,
  contentEditable,
  setContentEditable,
  dir,
  setDir,
}
/* include Not a pure module */
